crates/derive/src/pipeline/builder.rs

```diff
@@ -127,7 +136,9 @@ where
         let l2_chain_provider = builder.l2_chain_provider.expect("chain_provider must be set");
         let dap_source = builder.dap_source.expect("dap_source must be set");
         let attributes_builder = builder.builder.expect("builder must be set");
+        let metrics = builder.metrics.expect("metrics must be set");
 
+        // todo: add metrics to the stages
```

Comment:
Preference to create a sub-ticket in github issues over leaving TODO comments in the codebase.